### ISSUE TITLE
Slow SQL query for select worst measure

### ISSUE THREAD

**Issue by sebbacon** [ebmdatalab/openprescribing#1325]
From the postgres log:
`EXPLAIN ANALYZE` [here](https://explain.depesz.com/s/B2v6)

```sql
SELECT
  "frontend_measurevalue"."measure_id",
  AVG("frontend_measurevalue"."percentile") AS "average_percentile"
FROM
  "frontend_measurevalue"
INNER JOIN
  "frontend_measure"
ON
  ("frontend_measurevalue"."measure_id" = "frontend_measure"."id")
WHERE
  ("frontend_measurevalue"."month" >= '2018-04-01'::date
    AND "frontend_measurevalue"."percentile" IS NOT NULL
    AND "frontend_measurevalue"."practice_id" = 'F85052'
    AND "frontend_measure"."tags" @> ARRAY['core']::varchar(30)[]
    AND NOT ("frontend_measurevalue"."measure_id" = 'lpzomnibus'))
GROUP BY
  "frontend_measurevalue"."measure_id"
ORDER BY
  "average_percentile" DESC
LIMIT
  1```

**Comment by sebbacon** [comment]
This scans the entire `frontend_measurevalue` table, which apparently has NO indexes?!

The above query would basically be fixed by indexes on measure_id, practice_id and ccg_id.

The bigger question is... why are there no indexes currently?  Has something been lost somewhere along the way?

**Comment by sebbacon** [comment]
Possibly related to that incident a few months ago when I accidentally deleted some stuff via `CASCADE`?

**Comment by sebbacon** [comment]
This is fixed now by my adding indexes, but leaving the issue open to see if anyone can solve the mystery

**Comment by sebbacon** [comment]
This just happened again (see linked issue). 

Looking for this in the slow query logs (with `grep "LIMIT 1" postgresql-11-main.log.1`), the query has been slow since 11 March, when postgres was upgraded to version 11.

Adding the following indexes:

```sql
CREATE INDEX ON frontend_measurevalue (practice_id);
CREATE INDEX ON frontend_measurevalue (ccg_id);
CREATE INDEX ON frontend_measurevalue (tags);
CREATE INDEX ON frontend_measurevalue (month);
```

Why did this happen?  Data points:

* When this happened in Jan, there had not been a database upgrade
* This is the only table affected: every other table seems to have its indexes
* There was a data import this weekend; but it seems the Jan one was after the last time the indexes disappeared
* This table has had a migration applied to it (STPs etc) but I think that happened a while ago

**Comment by evansd** [comment]
I wondered if somehow a migration had ended up deleting these. The following shell pipeline dumps the SQL for all migrations:
```bash
find . -name '*.py' \
  | grep -P '/migrations/\d' \
  | sed -r 's#^.*/(\w+)/migrations/(.+)\.py$#\1 \2#g' \
  | sort \
  | xargs -n 2 -I % bash -c \
  'echo "-- %"; ./manage.py sqlmigrate %; echo -e "\n\n\n"' \
  | tee all_migrations.sql
```
Which results in [this SQL](https://gist.github.com/evansd/e5cd6eee0b014d2b0f17a535ee43de32).

This doesn't contain any `DROP INDEX` statements, and nothing else that looks suspicious touching `frontend_measurevalue`.

**Comment by sebbacon** [comment]
@inglesp this is just assigned to you to see if you have any idea why the indexes might have been dropped. If not we should probably just Move On because it is One of Those Things

**Comment by inglesp** [comment]
No idea.

**Comment by sebbacon** [comment]
This happened again during this month's import, so it's definitely not random...

**Comment by evansd** [comment]
Fixed by #1774
